packages/forms/src/styled/common/StyledLegend.ts

```diff
@@ -0,0 +1,31 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import styled from 'styled-components';
+import { hideVisually } from 'polished';
+import { retrieveComponentStyles, DEFAULT_THEME } from '@zendeskgarden/react-theming';
+
+const COMPONENT_ID = 'forms.fieldset_legend';
+
+interface IStyledLegend {
+  isCompact?: boolean;
+  isHidden?: boolean;
+}
+
+export const StyledLegend = styled.legend.attrs({
+  'data-garden-id': COMPONENT_ID,
+  'data-garden-version': PACKAGE_VERSION
+})<IStyledLegend>`
+  margin-bottom: ${props =>
+    props.isCompact ? props.theme.space.base : props.theme.space.base * 2}px;
+  ${props => props.isHidden && hideVisually()}
+  ${props => retrieveComponentStyles(COMPONENT_ID, props)};
+`;
```

Comment:
This should be (in abbreviated format):

```js
/**
 * 1. legend element reset.
 * /
StyledLegend = styled(StyledLabel).attrs({
  as: 'legend',
  ...
})`
  display: block;
  margin-bottom: ${props => props.theme.space.base * (props.isCompact ? 1 : 2)}px;
  padding: 0; /* [1] */

  ${props => retrieveComponentStyles(COMPONENT_ID, props)};
`;
```

Picks up all the missing styles, resets appropriately, and applies `hidden` as expected.